debezium-core/src/main/java/io/debezium/heartbeat/HeartbeatFactory.java

```diff
@@ -50,6 +58,11 @@ public HeartbeatFactory(CommonConnectorConfig connectorConfig, TopicNamingStrate
         this.schemaNameAdjuster = schemaNameAdjuster;
         this.connectionProvider = connectionProvider;
         this.errorHandler = errorHandler;
+        this.heartbeatFactories = new ArrayList<>(externalHeartbeatFactory
+                .stream()
+                .map(ServiceLoader.Provider::get)
+                .toList());
+        this.heartbeatFactories.add(new DatabaseHeartbeatFactory());
```

Comment:
Should we prioritize our implementation (the connector level one) before external ones?

Reply:
yeah, I agree